src/rpc/blockchain.cpp

```diff
@@ -1161,6 +1163,197 @@ static RPCHelpMan pruneblockchain()
     };
 }
 
+static RPCHelpMan dumpcoinstats()
+{
+    return RPCHelpMan{"dumpcoinstats",
+        "Dumps content of coinstats index to a CSV file. The resulting file contains the following columns:\n"
+        "- height\n"
+        "- block_hash\n"
+        "- timestamp\n"
+        "- mediantime\n"
+        "- version\n"
+        "- (total_)txs - Number of transactions included\n"
+        "- (total_)txouts - Number of transaction outputs created\n"
+        "- (total_)amount_spent - Amount of inputs spent\n"
+        "- (total_)amount_created - Amount of coins added to the UTXO set\n"
+        "- (total_)coinbase_amount - Coinbase subsidy amount\n"
+        "- (total_)new_outputs_ex_coinbase_amount - Amount of new outputs created by this block\n"
+        "- (total_)subsidy - Total subsidy paid out\n"
```

Comment:
Hehe, we are really getting into the weeds here :)

Yeah, the "paid out" part is definitely wrong. It wasn't paid out but also it can never be paid out in the the future, I still find this hard to express concisely so changed to "Maximum subsidy".

For the `unspendable_amount` comment: that is not a bug by my definition. When I wrote the coinstatsindex one of my goals was to make these non-existent amounts visible, and I summed this up with the other categories of lost coins under the umbrella term unspendables . See this code block in coinstatsindex:

```
    const CAmount unclaimed_rewards{(m_total_prevout_spent_amount + m_total_subsidy) - (m_total_new_outputs_ex_coinbase_amount + m_total_coinbase_amount + m_total_unspendable_amount)};
    m_total_unspendable_amount += unclaimed_rewards;
    m_total_unspendables_unclaimed_rewards += unclaimed_rewards;
```

If the definition for anything "unspendable" was that it needed have to existed before as an output in a tx, then yes, the unclaimed rewards should not be part of this category. But I think this is confusing to some without deep knowledge of how the subsidies work. They might think that these amounts could still be claimed some time in the future, since they haven't existed yet and there will be 21M, right? So I was more looking at this from the side of the 21M: for example the 12.5 are part of the 21M total (and part of the >19M that are considered mined) even though they will never exist because the moment to claim them has passed. Thus, I consider them unspendable even though they never even existed in the first place, i.e. "unspendable" includes "never spendable in the future". If we don't keep track of these to some degree we can never really do the full accounting that this export was intended for.

Reply:
I see, in that case just document that "unspendable" means "unspendable coins plus unclaimed subsidies".

Though I think it's cleaner to track `unspendable` and `unclaimed_rewards` separately, and then explain in the docs that "unclaimed" means it can never be claimed. I suspect anyone using an RPC like this will understand that.

Reply:
Done, amended the doc for the `unspendable_amount`.

I see your reasoning but at least for now let's keep it consistent with the index code and `gettxoutsetinfo` where the unclaimed rewards are in the unspendable 'category'.